### PR TITLE
use biome instead of prettier

### LINKED ISSUE #492: Use BiomeJS instead of prettier
### Preliminary Checks

- [X] I have reviewed https://docs.openstatus.dev/ for existing features that would solve my problem

- [X] I have searched for existing feature requests: https://github.com/openstatusHQ/openstatus/issues

- [X] This issue is not a question, general help request, or anything other than a feature request directly related to OpenStatus. Please ask questions in our Discord community: https://discord.com/invite/dHD4JtSfsn.

### Description

Switch from prettier to biomejs 

We should use the same config, also set up vscode with biomejs

### PR DESCRIPTION
**Pull request by gulfaraz**
resolves #492 

search for `biome-ignore` to address migration issues

### CODE REVIEW

File: apps/web/src/app/app/[workspaceSlug]/(dashboard)/settings/appearance/page.tsx [hunk 1]
```diff
@@ -12,18 +12,21 @@ export default function AppearancePage() {
 
   return (
     <div className="grid grid-cols-1 gap-4 sm:grid-cols-2 md:grid-cols-4">
+      {/* biome-ignore lint/a11y/useButtonType: */}
       <button onClick={() => setTheme("light")}>
```
Comment (mxkaske) [inline apps/web/src/app/app/[workspaceSlug]/(dashboard)/settings/appearance/page.tsx:16]: Interesting - wasn't expecting that you should [always specify](https://www.w3schools.com/jsref/prop_pushbutton_type.asp) the button type.

**Review by mxkaske — COMMENTED** [review]
Cool! Didn't know about all the `biome` recommendations but I really like comes with zero config and is biased.

File: apps/web/src/app/app/[workspaceSlug]/(dashboard)/settings/appearance/page.tsx (hunk 1 above)
Comment (gulfaraz) [reply apps/web/src/app/app/[workspaceSlug]/(dashboard)/settings/appearance/page.tsx:16]: Neither did I. This PR was a nice revisit to the basics.

**Comment by gulfaraz** [comment]
> Cool! Didn't know about all the `biome` recommendations but I really like comes with zero config and is biased.

I did make some config choices as specified in [biome.json](https://github.com/openstatusHQ/openstatus/pull/561/files#diff-2bc8a1f5e9380d5a187a4e90f11b4dd36c3abad6aea44c84be354a4f44cdec55)

This makes biome ignore the `shadcn` files so it's easy to update them.

https://github.com/openstatusHQ/openstatus/blob/b4f722ebde2b53df5b6bd9ec819e26d96b499d27/biome.json#L3-L8

This is equivalent to `prettier-plugin-sort-imports`. It is disabled but you can choose to enable it. Change `false` to `true` and run `npm run format`.

https://github.com/openstatusHQ/openstatus/blob/b4f722ebde2b53df5b6bd9ec819e26d96b499d27/biome.json#L9-L11

This was `tab` by default but changed it to `space`.

https://github.com/openstatusHQ/openstatus/blob/b4f722ebde2b53df5b6bd9ec819e26d96b499d27/biome.json#L18-L20

This makes biome ignore all `.json` files. There were minor auto-fixable format changes. I disabled it coz `prettier` was ignoring json files via `**/*.{ts,tsx,md}`.

https://github.com/openstatusHQ/openstatus/blob/b4f722ebde2b53df5b6bd9ec819e26d96b499d27/biome.json#L21-L25

You can also [disable any rule if you prefer](https://biomejs.dev/linter/#disable-a-lint-rule).

**Comment by thibaultleouay** [comment]
@gulfaraz 
That's cool my only concern is the tailwind class sort which does not work at the moment with Biome

**Comment by gulfaraz** [comment]
@thibaultleouay 

Biome doesn't have tailwind sorting yet, but it [is in discussion](https://github.com/biomejs/biome/issues/1274). So you have a decision to make,
1. We can **merge** and continue using biome without tailwind sort. When sorting is released, update biome and `npm run format` to fix the unsorted classes.
2. We **close** this PR until sorting is released.

**Comment by thibaultleouay** [comment]
@gulfaraz  

I would love to use Biomejs, 
Can we keep it for code? And just have prettier at the same time but only for class sorting? Which will be removed when biomejs implements it?

What's your thought on it?

**Comment by gulfaraz** [comment]
> And just have prettier at the same time but only for class sorting?

[That is not possible.](https://github.com/prettier/prettier/issues/9171)

**Comment by thibaultleouay** [comment]
@gulfaraz  
if you don't mind waiting for couple of days until this is released 
https://github.com/biomejs/biome/issues/1274

**Comment by gulfaraz** [comment]
No problem.

**Comment by thibaultleouay** [comment]
I think they added some support to class sorting 
https://github.com/biomejs/biome/discussions/164 

Let's integrate biomejs 🔥

**Comment by gulfaraz** [comment]
The related PR https://github.com/biomejs/biome/pull/1362 (2 weeks ago) is not in their [latest release](https://github.com/biomejs/biome/releases/tag/cli%2Fv1.5.3) (3 weeks ago) yet.

**Comment by gulfaraz** [comment]
closed in favour of #728

**Comment by thibaultleouay** [comment]
@gulfaraz  Would love if you double check the other PR, it seems like a mess to me 😭

**Pull request closed without merging**